### ISSUE TITLE
Available channels for Tuya SmartBulbs

### ISSUE THREAD

**Issue by rlarranaga** [smarthomej/addons#410]
Hello,
I used Autodiscover to get my devices imported into openhab. I was surprised that none of my smart bulbs contains a default switch "on/off" channel.

I see that some of the other discovered channels (colour, brightness) have a second parameter "DP 2" that is supposed to work like an ON/OFF switch, however, looking at some of these parameters, de default value that I see is "20". I played with the lights to see if this parameter would change, but it did not.

In short, the question is, what is the best way to turn the lights on and off?
Thanks

**Comment by J-N-K** [comment]
As always in openHAB: send OFF or ON to the brightness/color channel. These channels accept ON/OFF, INCREASE/DECREASE and PercentType for brightness (and HSBType in case of a color channel)

**Comment by rlarranaga** [comment]
After playing around with the binding for a bit, let me rephrase the question:

I have tuya bulbs from 2 different vendors in my house. Each was discovered with a number of channels (control_data, scene_data, colour_data, bright_value, temp_value ,work_mode, music_data, countdown_1). 

I was surprised in that there was no on/off channel discovered automatically, even though both the bright_data and the colour_data channels hint at what DP the switch channel would use. Is there a reason why there isn't an On/Off channel added automatically to the bulbs? Specially taking into consideration that switching lights on and off is likely amongst the actions that are used the most with lightbulbs.

Is this the expected behavior, or is it something related with the specific metadata scheme of the 2 vendors I am using?
Happy to share the product IDs if needed

After adding on of channels, (Utilizing the hinted DP from colour_data and bright_value) everything looks great, so thanks for all the work on this binding!

**Comment by J-N-K** [comment]
It is an openHAB concept that a device that supports 

- only ON/OFF is modeled as a thing with a channel of type `Switch`
- absolute brightness values (`PercentType`) is modeled as a thing with a channel of type `Dimmer`(which also allows using ON/OFF)
- color values is modeled as a thing with a channel of type `Color` (which also allows using ON/OFF and absolute brightness values)

Since you can link a `Color` type (or `Dimmer` type) channel to a `Switch` item and it works as expected, there is no need for a `Switch` channel and therefore it is considered as duplicate and not added. This is not specific to this binding, it's a general rule. 

As a side-note: You can link channels to more than one item, so you can have a `Switch` item called `DiningRoom_Power` and a `Dimmer` item called `DiningRoom_Brightness` and link both to the same channel of the thing. 

You can instruct BasicUI (and I believe also MainUI, but not sure about that) to render the `Dimmer` item as a switch if you prefer that and from rules you can also send `ON` or `OFF` to a `Color`/`Dimmer` item, so in most cases a separate `Switch` item is not needed.

**Comment by rlarranaga** [comment]
Ok, thanks, that is an interesting concept I didn't know about. I will test using Sending On/Off commands straight to the brightness channel.

Appreciate the help!

**Comment by J-N-K** [comment]
Feel free to ask if there are any further questions.

**Comment by rlarranaga** [comment]
Hey,

After using the brightness channels as On/Off Switches for a couple of days, I had to revert to using the On/Of Channels that I created.

One of the issues I am facing is when using the light bulbs in groups. When I do this, I usually create an aggregation for the group that sets the group status as ON when all members are ON, and OFF in any other case.

The problem I encountered is that some of these lights will keep a brightness level of 100 when off. Since 100 appears to be equivalent to ON, this will make the group report the status incorrectly, which in turn will make the group based switches in hab panel non working.

Is this an expected behavior?

Thanks

**Comment by J-N-K** [comment]
Group aggregation is a little bit difficult if you link items that behave differently. Tuya devices show a very broad range of behavior, some report "zero" as brightness when turned off, some don't. Some report "off" if brightness is "0", some stay "on".

My best advice would be to have a rule that listens for the ON/OFF commands and properly updates the group state.

**Comment by rlarranaga** [comment]
In this case, I am using groups with the exact same models of lamps, which
is why I would have expected all lights to show the same state. Like, if i
am send all lights a command for 0 brightness, regardless of the final
value, they should all have the same one, shouldn't they?

Thanks
